pkg/standalone-utils/contracts/relayer/VaultActions.sol

```diff
@@ -239,7 +278,18 @@ abstract contract VaultActions is IBaseRelayerLibrary {
         if (kind == PoolKind.WEIGHTED) {
             return _doWeightedExitChainedReferenceReplacements(userData);
         } else {
-            _revert(Errors.UNHANDLED_EXIT_KIND);
+            // exitKind just looks at the first value, same for all, so we can use the current library function
+            uint8 exitKind = uint8(StablePoolUserData.exitKind(userData));
```

Comment:
I find this odd here, all we gain is to not have to make the call further down. I'd move this inside each of the `_doReplacements` functions to clarity, also making the structure of this `if-else` chain clearer.